src/main/java/org/spdx/sbom/gradle/maven/DependencyResolver.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 The Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.spdx.sbom.gradle.maven;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.artifacts.result.DependencyResult;
+import org.gradle.api.artifacts.result.ResolvedArtifactResult;
+import org.gradle.api.artifacts.result.ResolvedComponentResult;
+import org.gradle.api.artifacts.result.ResolvedDependencyResult;
+import org.gradle.maven.MavenModule;
+import org.gradle.maven.MavenPomArtifact;
+
+public class DependencyResolver {
+  private final DependencyHandler dependencies;
+
+  public static DependencyResolver newDependencyResolver(DependencyHandler dependencies) {
+    return new DependencyResolver(dependencies);
+  }
+
+  DependencyResolver(DependencyHandler dependencies) {
+    this.dependencies = dependencies;
+  }
+
+  @SuppressWarnings("unchecked")
+  public List<ResolvedArtifactResult> resolveDependencies(
+      List<ResolvedComponentResult> rootComponents) {
+    Set<ComponentIdentifier> componentIds = gatherSelectedDependencies(rootComponents);
+
+    return dependencies
+        .createArtifactResolutionQuery()
+        .forComponents(componentIds)
+        .withArtifacts(MavenModule.class, MavenPomArtifact.class)
+        .execute()
+        .getResolvedComponents()
+        .stream()
+        .flatMap(
```

Comment:
Ah so actually this creates an issue, as our resolved artifacts now only references POMs. So the result provided to https://github.com/spdx/spdx-gradle-plugin/pull/72/files#diff-02bb30d3890899ff394c66a6e22f3a5857c39ae022a1a696a25e5958925d2ed5R137 is only poms instead of the jar files.

Reply:
See output doc:
```diff
{
    "SPDXID" : "SPDXRef-gnrtd3",
    "checksums" : [ {
      "algorithm" : "SHA256",
      "checksumValue" : "9f26df47de9b3a2619c73e62f82477ba71c02a7a3d61ce22e1e338e6829ab6f9"
    }, {
      "algorithm" : "SHA1",
      "checksumValue" : "15fc7076dc38c66850b11413d7d346621655cf46"
    } ],
    "copyrightText" : "NOASSERTION",
-    "downloadLocation" : "https://repo.maven.apache.org/maven2/dev/sigstore/sigstore-java/0.3.0/sigstore-java-0.3.0.jar",
+    "downloadLocation" : "https://repo.maven.apache.org/maven2/dev/sigstore/sigstore-java/0.3.0/sigstore-java-0.3.0.pom",
    "externalRefs" : [ {
      "referenceCategory" : "PACKAGE-MANAGER",
      "referenceLocator" : "pkg:maven/dev.sigstore/sigstore-java@0.3.0?repository_url=repo.maven.apache.org%2Fmaven2",
      "referenceType" : "purl"
    } ],
    "filesAnalyzed" : false,
    "licenseConcluded" : "NOASSERTION",
    "licenseDeclared" : "Apache-2.0",
    "licenseInfoFromFiles" : [ ],
    "name" : "dev.sigstore:sigstore-java:0.3.0",
    "supplier" : "Organization: Sigstore",
    "versionInfo" : "0.3.0"
  }
  ```

Reply:
Gonna try to figure this one out later tonight or tomorrow.

Reply:
I think filtering on componentarrifactresult outside of this method in the plugin. We can use mavenpomartifact for the pom resolver and a mavenmodule filter for "resolvedartifacts"

Reply:
I don't think it's possible to get the default artifact from these APIs ([although I asked for help with that](https://gradle-community.slack.com/archives/CAHSN3LDN/p1699854619838439)), so the approach I took for now is to follow the [Maven naming conventions](https://maven.apache.org/guides/mini/guide-naming-conventions.html) and use the artifactId + the version to determine the filename.

Reply:
Got confirmation about using ARQ for normal artifacts:

> You cannot use an ARQ for resolving normal artifacts. They are only for POM, source artifact and javadoc artifact.